calculator/dom.js

```diff
@@ -0,0 +1,160 @@
+let buttons = document.querySelectorAll('.button'),
+    numbers = document.querySelectorAll('.number'),
+    operations = document.querySelectorAll('.operation'),
+    decimal = document.getElementById('decimal'),
+    cleaner_all = document.getElementById('ac'),
+    cleaner_window = document.getElementById('c'),
+    display = document.getElementById('display'),
+    sign = document.getElementById('sign'),
+    memoryCurrentNumber = 0,
+    memoryNewNumber = false,
+    memoryError = false,
+    memoryPendingOperation = '';
+
+for (let i = 0; i < numbers.length; i++) {
+    let number = numbers[i];
+    number.addEventListener('click', (e) => {
+        if (!memoryError) {
+            pressNummber(e.target.textContent);
+        }
+    });
+};
+
+for (let i = 0; i < operations.length; i++) {
+    let operation = operations[i];
+    operation.addEventListener('click', (e) => {
+        if (!memoryError) {
+            addOperation(e.target.textContent);
+        }
+    });
+};
+
+sign.addEventListener('click', () => {
+    if (!memoryError) {
+        changeSign();
+    }
+})
+
+decimal.addEventListener('click', () => {
+    if (!memoryError) {
+        addDecimal();
+    }
+});
+
+cleaner_all.addEventListener('click', () => {
+    clearAll();
+})
+
+cleaner_window.addEventListener('click', () => {
+    if (!memoryError) {
+        clearWindow();
+    }
+})
+
+function pressNummber(number) {
+    if (memoryNewNumber && display.value === '-0') {
+        display.value = '-' + number;
+        memoryNewNumber = false;
+    }
+    else if (memoryNewNumber) {
+        display.value = number;
+        memoryNewNumber = false;
+    }
+    else {
+        if (display.value === '0') {
+            display.value = number;
+        }
+        else if (display.value === '-0') {
+            display.value = '-' + number;
+        }
+        else {
+            display.value += number;
+        }
+    }
+    
+};
+
+function addOperation(oper) {
```

Comment:
this is a very hard to read function with very high cyclomatic complexity. try to get rid of nested control control flow operators